src/Compilers/Core/Portable/MetadataReader/MetadataTypeName.cs

```diff
@@ -195,6 +229,34 @@ public string TypeName
             }
         }
 
+        public ReadOnlyMemory<char> NamespaceNameMemory
+        {
+            get
+            {
+                if (_namespaceNameMemory.Equals(default))
+                {
+                    Debug.Assert(_fullName != null);
+                    _typeNameMemory = MetadataHelpers.SplitQualifiedName(_fullName, out _namespaceNameMemory);
```

Comment:
>MetadataHelpers.SplitQualifiedName [](http://example.com/codeflow?start=38&length=34)

Should probably avoid doing the work when ```_namespaceName``` has been calculated already. And vise versa.

Reply:
We probably should always do the work in terms of ```ReadOnlyMemory<char>``` and then realizing strings, or reusing existing strings when possible.

Reply:
And reusing ```s_splitQualifiedNameSystem``` as well.